### PR TITLE
Adjusted Time __repr__ to correctly handle long lists of Times

### LINKED ISSUE #9384: String representation for Time can be too long
### Description

The ``__repr__`` for ``Time`` seems to show all values even for large arrays.

[EDIT from @taldcroft]

After an attempt to address this in #9525 by hardcoding to a smaller limit of 50, it became apparent that this introduces new complications in the user interface (specifically how to override that value of 50 in a consistent way).

At the end of the day I think the best answer is to document how to control the repr formatting via `np.printoptions`.  This way the `Time` object behaves more like a numpy array and there is consistency in the repr API.
```
In [15]: t = Time(np.arange(1000), format='cxcsec')

In [16]: with np.printoptions(threshold=10):
    ...:     print(repr(t))
    ...:     
<Time object: scale='tt' format='cxcsec' value=[  0.   1.   2. ... 997. 998. 999.]>
```

This now turns into an easy issue that somebody is invited to fix with a documentation example.

[END EDIT]

### Expected behavior

```
In [1]: import numpy as np                                                                                                                             

In [2]: from astropy.time import Time                                                                                                                  

In [3]: import numpy as np 
   ...: t4 = Time(np.linspace(2458780, 2458790, 1000), format='jd') 
   ...: t4                                                                                                                                             
Out[3]: 
<Time object: scale='utc' format='jd' value=[2458780.         2458780.01001001 ... 2458789.98998999 2458790.        ]>
```

### Actual behavior

```
In [1]: import numpy as np                                                                                                                             

In [2]: from astropy.time import Time                                                                                                                  

In [3]: import numpy as np 
   ...: t4 = Time(np.linspace(2458780, 2458790, 1000), format='jd') 
   ...: t4                                                                                                                                             
Out[3]: 
<Time object: scale='utc' format='jd' value=[2458780.         2458780.01001001 2458780.02002002 2458780.03003003
 2458780.04004004 2458780.05005005 2458780.06006006 2458780.07007007
 2458780.08008008 2458780.09009009 2458780.1001001  2458780.11011011
 2458780.12012012 2458780.13013013 2458780.14014014 2458780.15015015
 2458780.16016016 2458780.17017017 2458780.18018018 2458780.19019019
 2458780.2002002  2458780.21021021 2458780.22022022 2458780.23023023
 2458780.24024024 2458780.25025025 2458780.26026026 2458780.27027027
 2458780.28028028 2458780.29029029 2458780.3003003  2458780.31031031
 2458780.32032032 2458780.33033033 2458780.34034034 2458780.35035035
 2458780.36036036 2458780.37037037 2458780.38038038 2458780.39039039
 2458780.4004004  2458780.41041041 2458780.42042042 2458780.43043043
 2458780.44044044 2458780.45045045 2458780.46046046 2458780.47047047
 2458780.48048048 2458780.49049049 2458780.5005005  2458780.51051051
 2458780.52052052 2458780.53053053 2458780.54054054 2458780.55055055
 2458780.56056056 2458780.57057057 2458780.58058058 2458780.59059059
 2458780.6006006  2458780.61061061 2458780.62062062 2458780.63063063
 2458780.64064064 2458780.65065065 2458780.66066066 2458780.67067067
 2458780.68068068 2458780.69069069 2458780.7007007  2458780.71071071
 2458780.72072072 2458780.73073073 2458780.74074074 2458780.75075075
 2458780.76076076 2458780.77077077 2458780.78078078 2458780.79079079
 2458780.8008008  2458780.81081081 2458780.82082082 2458780.83083083
 2458780.84084084 2458780.85085085 2458780.86086086 2458780.87087087
 2458780.88088088 2458780.89089089 2458780.9009009  2458780.91091091
 2458780.92092092 2458780.93093093 2458780.94094094 2458780.95095095
 2458780.96096096 2458780.97097097 2458780.98098098 2458780.99099099
 2458781.001001   2458781.01101101 2458781.02102102 2458781.03103103
 2458781.04104104 2458781.05105105 2458781.06106106 2458781.07107107
 2458781.08108108 2458781.09109109 2458781.1011011  2458781.11111111
 2458781.12112112 2458781.13113113 2458781.14114114 2458781.15115115
 2458781.16116116 2458781.17117117 2458781.18118118 2458781.19119119
 2458781.2012012  2458781.21121121 2458781.22122122 2458781.23123123
 2458781.24124124 2458781.25125125 2458781.26126126 2458781.27127127
 2458781.28128128 2458781.29129129 2458781.3013013  2458781.31131131
 2458781.32132132 2458781.33133133 2458781.34134134 2458781.35135135
 2458781.36136136 2458781.37137137 2458781.38138138 2458781.39139139
 2458781.4014014  2458781.41141141 2458781.42142142 2458781.43143143
 2458781.44144144 2458781.45145145 2458781.46146146 2458781.47147147
 2458781.48148148 2458781.49149149 2458781.5015015  2458781.51151151
 2458781.52152152 2458781.53153153 2458781.54154154 2458781.55155155
 2458781.56156156 2458781.57157157 2458781.58158158 2458781.59159159
 2458781.6016016  2458781.61161161 2458781.62162162 2458781.63163163
 2458781.64164164 2458781.65165165 2458781.66166166 2458781.67167167
 2458781.68168168 2458781.69169169 2458781.7017017  2458781.71171171
 2458781.72172172 2458781.73173173 2458781.74174174 2458781.75175175
 2458781.76176176 2458781.77177177 2458781.78178178 2458781.79179179
 2458781.8018018  2458781.81181181 2458781.82182182 2458781.83183183
 2458781.84184184 2458781.85185185 2458781.86186186 2458781.87187187
 2458781.88188188 2458781.89189189 2458781.9019019  2458781.91191191
 2458781.92192192 2458781.93193193 2458781.94194194 2458781.95195195
 2458781.96196196 2458781.97197197 2458781.98198198 2458781.99199199
 2458782.002002   2458782.01201201 2458782.02202202 2458782.03203203
 2458782.04204204 2458782.05205205 2458782.06206206 2458782.07207207
 2458782.08208208 2458782.09209209 2458782.1021021  2458782.11211211
 2458782.12212212 2458782.13213213 2458782.14214214 2458782.15215215
 2458782.16216216 2458782.17217217 2458782.18218218 2458782.19219219
 2458782.2022022  2458782.21221221 2458782.22222222 2458782.23223223
 2458782.24224224 2458782.25225225 2458782.26226226 2458782.27227227
 2458782.28228228 2458782.29229229 2458782.3023023  2458782.31231231
 2458782.32232232 2458782.33233233 2458782.34234234 2458782.35235235
 2458782.36236236 2458782.37237237 2458782.38238238 2458782.39239239
 2458782.4024024  2458782.41241241 2458782.42242242 2458782.43243243
 2458782.44244244 2458782.45245245 2458782.46246246 2458782.47247247
 2458782.48248248 2458782.49249249 2458782.5025025  2458782.51251251
 2458782.52252252 2458782.53253253 2458782.54254254 2458782.55255255
 2458782.56256256 2458782.57257257 2458782.58258258 2458782.59259259
 2458782.6026026  2458782.61261261 2458782.62262262 2458782.63263263
 2458782.64264264 2458782.65265265 2458782.66266266 2458782.67267267
 2458782.68268268 2458782.69269269 2458782.7027027  2458782.71271271
 2458782.72272272 2458782.73273273 2458782.74274274 2458782.75275275
 2458782.76276276 2458782.77277277 2458782.78278278 2458782.79279279
 2458782.8028028  2458782.81281281 2458782.82282282 2458782.83283283
 2458782.84284284 2458782.85285285 2458782.86286286 2458782.87287287
 2458782.88288288 2458782.89289289 2458782.9029029  2458782.91291291
 2458782.92292292 2458782.93293293 2458782.94294294 2458782.95295295
 2458782.96296296 2458782.97297297 2458782.98298298 2458782.99299299
 2458783.003003   2458783.01301301 2458783.02302302 2458783.03303303
 2458783.04304304 2458783.05305305 2458783.06306306 2458783.07307307
 2458783.08308308 2458783.09309309 2458783.1031031  2458783.11311311
 2458783.12312312 2458783.13313313 2458783.14314314 2458783.15315315
 2458783.16316316 2458783.17317317 2458783.18318318 2458783.19319319
 2458783.2032032  2458783.21321321 2458783.22322322 2458783.23323323
 2458783.24324324 2458783.25325325 2458783.26326326 2458783.27327327
 2458783.28328328 2458783.29329329 2458783.3033033  2458783.31331331
 2458783.32332332 2458783.33333333 2458783.34334334 2458783.35335335
 2458783.36336336 2458783.37337337 2458783.38338338 2458783.39339339
 2458783.4034034  2458783.41341341 2458783.42342342 2458783.43343343
 2458783.44344344 2458783.45345345 2458783.46346346 2458783.47347347
 2458783.48348348 2458783.49349349 2458783.5035035  2458783.51351351
 2458783.52352352 2458783.53353353 2458783.54354354 2458783.55355355
 2458783.56356356 2458783.57357357 2458783.58358358 2458783.59359359
 2458783.6036036  2458783.61361361 2458783.62362362 2458783.63363363
 2458783.64364364 2458783.65365365 2458783.66366366 2458783.67367367
 2458783.68368368 2458783.69369369 2458783.7037037  2458783.71371371
 2458783.72372372 2458783.73373373 2458783.74374374 2458783.75375375
 2458783.76376376 2458783.77377377 2458783.78378378 2458783.79379379
 2458783.8038038  2458783.81381381 2458783.82382382 2458783.83383383
 2458783.84384384 2458783.85385385 2458783.86386386 2458783.87387387
 2458783.88388388 2458783.89389389 2458783.9039039  2458783.91391391
 2458783.92392392 2458783.93393393 2458783.94394394 2458783.95395395
 2458783.96396396 2458783.97397397 2458783.98398398 2458783.99399399
 2458784.004004   2458784.01401401 2458784.02402402 2458784.03403403
 2458784.04404404 2458784.05405405 2458784.06406406 2458784.07407407
 2458784.08408408 2458784.09409409 2458784.1041041  2458784.11411411
 2458784.12412412 2458784.13413413 2458784.14414414 2458784.15415415
 2458784.16416416 2458784.17417417 2458784.18418418 2458784.19419419
 2458784.2042042  2458784.21421421 2458784.22422422 2458784.23423423
 2458784.24424424 2458784.25425425 2458784.26426426 2458784.27427427
 2458784.28428428 2458784.29429429 2458784.3043043  2458784.31431431
 2458784.32432432 2458784.33433433 2458784.34434434 2458784.35435435
 2458784.36436436 2458784.37437437 2458784.38438438 2458784.39439439
 2458784.4044044  2458784.41441441 2458784.42442442 2458784.43443443
 2458784.44444444 2458784.45445445 2458784.46446446 2458784.47447447
 2458784.48448448 2458784.49449449 2458784.5045045  2458784.51451451
 2458784.52452452 2458784.53453453 2458784.54454454 2458784.55455455
 2458784.56456456 2458784.57457457 2458784.58458458 2458784.59459459
 2458784.6046046  2458784.61461461 2458784.62462462 2458784.63463463
 2458784.64464464 2458784.65465465 2458784.66466466 2458784.67467467
 2458784.68468468 2458784.69469469 2458784.7047047  2458784.71471471
 2458784.72472472 2458784.73473473 2458784.74474474 2458784.75475475
 2458784.76476476 2458784.77477477 2458784.78478479 2458784.79479479
 2458784.8048048  2458784.81481481 2458784.82482482 2458784.83483484
 2458784.84484484 2458784.85485485 2458784.86486486 2458784.87487487
 2458784.88488489 2458784.89489489 2458784.9049049  2458784.91491491
 2458784.92492492 2458784.93493494 2458784.94494495 2458784.95495495
 2458784.96496496 2458784.97497497 2458784.98498499 2458784.994995
 2458785.005005   2458785.01501501 2458785.02502503 2458785.03503504
 2458785.04504505 2458785.05505505 2458785.06506506 2458785.07507508
 2458785.08508509 2458785.0950951  2458785.10510511 2458785.11511511
 2458785.12512513 2458785.13513514 2458785.14514515 2458785.15515516
 2458785.16516516 2458785.17517518 2458785.18518519 2458785.1951952
 2458785.20520521 2458785.21521521 2458785.22522523 2458785.23523524
 2458785.24524525 2458785.25525526 2458785.26526527 2458785.27527528
 2458785.28528529 2458785.2952953  2458785.30530531 2458785.31531532
 2458785.32532533 2458785.33533534 2458785.34534535 2458785.35535536
 2458785.36536537 2458785.37537538 2458785.38538539 2458785.3953954
 2458785.40540541 2458785.41541542 2458785.42542543 2458785.43543544
 2458785.44544545 2458785.45545546 2458785.46546547 2458785.47547548
 2458785.48548549 2458785.4954955  2458785.50550551 2458785.51551552
 2458785.52552553 2458785.53553554 2458785.54554555 2458785.55555556
 2458785.56556557 2458785.57557558 2458785.58558559 2458785.5955956
 2458785.60560561 2458785.61561562 2458785.62562563 2458785.63563564
 2458785.64564565 2458785.65565566 2458785.66566567 2458785.67567568
 2458785.68568569 2458785.6956957  2458785.70570571 2458785.71571572
 2458785.72572573 2458785.73573574 2458785.74574575 2458785.75575576
 2458785.76576577 2458785.77577578 2458785.78578579 2458785.7957958
 2458785.80580581 2458785.81581582 2458785.82582583 2458785.83583584
 2458785.84584585 2458785.85585586 2458785.86586587 2458785.87587588
 2458785.88588589 2458785.8958959  2458785.90590591 2458785.91591592
 2458785.92592593 2458785.93593594 2458785.94594595 2458785.95595596
 2458785.96596597 2458785.97597598 2458785.98598599 2458785.995996
 2458786.00600601 2458786.01601602 2458786.02602603 2458786.03603604
 2458786.04604605 2458786.05605606 2458786.06606607 2458786.07607608
 2458786.08608609 2458786.0960961  2458786.10610611 2458786.11611612
 2458786.12612613 2458786.13613614 2458786.14614615 2458786.15615616
 2458786.16616617 2458786.17617618 2458786.18618619 2458786.1961962
 2458786.20620621 2458786.21621622 2458786.22622623 2458786.23623624
 2458786.24624625 2458786.25625626 2458786.26626627 2458786.27627628
 2458786.28628629 2458786.2962963  2458786.30630631 2458786.31631632
 2458786.32632633 2458786.33633634 2458786.34634635 2458786.35635636
 2458786.36636637 2458786.37637638 2458786.38638639 2458786.3963964
 2458786.40640641 2458786.41641642 2458786.42642643 2458786.43643644
 2458786.44644645 2458786.45645646 2458786.46646647 2458786.47647648
 2458786.48648649 2458786.4964965  2458786.50650651 2458786.51651652
 2458786.52652653 2458786.53653654 2458786.54654655 2458786.55655656
 2458786.56656657 2458786.57657658 2458786.58658659 2458786.5965966
 2458786.60660661 2458786.61661662 2458786.62662663 2458786.63663664
 2458786.64664665 2458786.65665666 2458786.66666667 2458786.67667668
 2458786.68668669 2458786.6966967  2458786.70670671 2458786.71671672
 2458786.72672673 2458786.73673674 2458786.74674675 2458786.75675676
 2458786.76676677 2458786.77677678 2458786.78678679 2458786.7967968
 2458786.80680681 2458786.81681682 2458786.82682683 2458786.83683684
 2458786.84684685 2458786.85685686 2458786.86686687 2458786.87687688
 2458786.88688689 2458786.8968969  2458786.90690691 2458786.91691692
 2458786.92692693 2458786.93693694 2458786.94694695 2458786.95695696
 2458786.96696697 2458786.97697698 2458786.98698699 2458786.996997
 2458787.00700701 2458787.01701702 2458787.02702703 2458787.03703704
 2458787.04704705 2458787.05705706 2458787.06706707 2458787.07707708
 2458787.08708709 2458787.0970971  2458787.10710711 2458787.11711712
 2458787.12712713 2458787.13713714 2458787.14714715 2458787.15715716
 2458787.16716717 2458787.17717718 2458787.18718719 2458787.1971972
 2458787.20720721 2458787.21721722 2458787.22722723 2458787.23723724
 2458787.24724725 2458787.25725726 2458787.26726727 2458787.27727728
 2458787.28728729 2458787.2972973  2458787.30730731 2458787.31731732
 2458787.32732733 2458787.33733734 2458787.34734735 2458787.35735736
 2458787.36736737 2458787.37737738 2458787.38738739 2458787.3973974
 2458787.40740741 2458787.41741742 2458787.42742743 2458787.43743744
 2458787.44744745 2458787.45745746 2458787.46746747 2458787.47747748
 2458787.48748749 2458787.4974975  2458787.50750751 2458787.51751752
 2458787.52752753 2458787.53753754 2458787.54754755 2458787.55755756
 2458787.56756757 2458787.57757758 2458787.58758759 2458787.5975976
 2458787.60760761 2458787.61761762 2458787.62762763 2458787.63763764
 2458787.64764765 2458787.65765766 2458787.66766767 2458787.67767768
 2458787.68768769 2458787.6976977  2458787.70770771 2458787.71771772
 2458787.72772773 2458787.73773774 2458787.74774775 2458787.75775776
 2458787.76776777 2458787.77777778 2458787.78778779 2458787.7977978
 2458787.80780781 2458787.81781782 2458787.82782783 2458787.83783784
 2458787.84784785 2458787.85785786 2458787.86786787 2458787.87787788
 2458787.88788789 2458787.8978979  2458787.90790791 2458787.91791792
 2458787.92792793 2458787.93793794 2458787.94794795 2458787.95795796
 2458787.96796797 2458787.97797798 2458787.98798799 2458787.997998
 2458788.00800801 2458788.01801802 2458788.02802803 2458788.03803804
 2458788.04804805 2458788.05805806 2458788.06806807 2458788.07807808
 2458788.08808809 2458788.0980981  2458788.10810811 2458788.11811812
 2458788.12812813 2458788.13813814 2458788.14814815 2458788.15815816
 2458788.16816817 2458788.17817818 2458788.18818819 2458788.1981982
 2458788.20820821 2458788.21821822 2458788.22822823 2458788.23823824
 2458788.24824825 2458788.25825826 2458788.26826827 2458788.27827828
 2458788.28828829 2458788.2982983  2458788.30830831 2458788.31831832
 2458788.32832833 2458788.33833834 2458788.34834835 2458788.35835836
 2458788.36836837 2458788.37837838 2458788.38838839 2458788.3983984
 2458788.40840841 2458788.41841842 2458788.42842843 2458788.43843844
 2458788.44844845 2458788.45845846 2458788.46846847 2458788.47847848
 2458788.48848849 2458788.4984985  2458788.50850851 2458788.51851852
 2458788.52852853 2458788.53853854 2458788.54854855 2458788.55855856
 2458788.56856857 2458788.57857858 2458788.58858859 2458788.5985986
 2458788.60860861 2458788.61861862 2458788.62862863 2458788.63863864
 2458788.64864865 2458788.65865866 2458788.66866867 2458788.67867868
 2458788.68868869 2458788.6986987  2458788.70870871 2458788.71871872
 2458788.72872873 2458788.73873874 2458788.74874875 2458788.75875876
 2458788.76876877 2458788.77877878 2458788.78878879 2458788.7987988
 2458788.80880881 2458788.81881882 2458788.82882883 2458788.83883884
 2458788.84884885 2458788.85885886 2458788.86886887 2458788.87887888
 2458788.88888889 2458788.8988989  2458788.90890891 2458788.91891892
 2458788.92892893 2458788.93893894 2458788.94894895 2458788.95895896
 2458788.96896897 2458788.97897898 2458788.98898899 2458788.998999
 2458789.00900901 2458789.01901902 2458789.02902903 2458789.03903904
 2458789.04904905 2458789.05905906 2458789.06906907 2458789.07907908
 2458789.08908909 2458789.0990991  2458789.10910911 2458789.11911912
 2458789.12912913 2458789.13913914 2458789.14914915 2458789.15915916
 2458789.16916917 2458789.17917918 2458789.18918919 2458789.1991992
 2458789.20920921 2458789.21921922 2458789.22922923 2458789.23923924
 2458789.24924925 2458789.25925926 2458789.26926927 2458789.27927928
 2458789.28928929 2458789.2992993  2458789.30930931 2458789.31931932
 2458789.32932933 2458789.33933934 2458789.34934935 2458789.35935936
 2458789.36936937 2458789.37937938 2458789.38938939 2458789.3993994
 2458789.40940941 2458789.41941942 2458789.42942943 2458789.43943944
 2458789.44944945 2458789.45945946 2458789.46946947 2458789.47947948
 2458789.48948949 2458789.4994995  2458789.50950951 2458789.51951952
 2458789.52952953 2458789.53953954 2458789.54954955 2458789.55955956
 2458789.56956957 2458789.57957958 2458789.58958959 2458789.5995996
 2458789.60960961 2458789.61961962 2458789.62962963 2458789.63963964
 2458789.64964965 2458789.65965966 2458789.66966967 2458789.67967968
 2458789.68968969 2458789.6996997  2458789.70970971 2458789.71971972
 2458789.72972973 2458789.73973974 2458789.74974975 2458789.75975976
 2458789.76976977 2458789.77977978 2458789.78978979 2458789.7997998
 2458789.80980981 2458789.81981982 2458789.82982983 2458789.83983984
 2458789.84984985 2458789.85985986 2458789.86986987 2458789.87987988
 2458789.88988989 2458789.8998999  2458789.90990991 2458789.91991992
 2458789.92992993 2458789.93993994 2458789.94994995 2458789.95995996
 2458789.96996997 2458789.97997998 2458789.98998999 2458790.        ]>
```

### PR DESCRIPTION
**Pull request by colathro**
The repr response  has some test dependencies which needed to be updated also.

Please also help identify any edge cases I may miss here.

### Description

This pull request is to address incorrect repr method on Time.

Fixes #9384

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -242,6 +242,8 @@ astropy.time
   existing method for ``TimeDelta``) so that one can get values in a given
   ``format`` and possible ``subfmt`` (e.g., ``to_value('mjd', 'str')``. [#9361]
 
+- ``Time`` will correctly display __repr__() values. [#9525]
```
Comment (eteq) [inline CHANGES.rst]: You might want to add a bit more detail in this note to say that it's specifically about long lists, since "correctly" is a bit ambiguous.

**Comment by pllim** [comment]
Thank you. When the 4.1 section is available in the change log, you would need to move the entry there; but for now, you can ignore the "changelog" bot.

**Comment by mhvk** [comment]
Thanks for the PR! However, could you describe more clearly what you intend to achieve? E.g., I think part is just to ensure that if we display a scalar and an array that we use quotes the same way:
```
t = Time(50000, np.arange(9.), format='mjd')
t.format='iso'
t  # array of times uses quotes
<Time object: scale='utc' format='iso' value=['1995-10-10 00:00:00.000' '1995-10-11 00:00:00.000'
 '1995-10-12 00:00:00.000' '1995-10-13 00:00:00.000'
 '1995-10-14 00:00:00.000' '1995-10-15 00:00:00.000'
 '1995-10-16 00:00:00.000' '1995-10-17 00:00:00.000'
 '1995-10-18 00:00:00.000']>
t[0]  # scalar time does not show quotes
<Time object: scale='utc' format='iso' value=1995-10-10 00:00:00.000>
```
This makes sense - though I'm not sure I wouldn't go the other direction, and always use quotes!

But you also need to make sure that whatever you do works for multi-dimensional arrays as well.  E.g., continuing from the above,
```
t.reshape(3, 3)
<Time object: scale='utc' format='iso' value=[['1995-10-10 00:00:00.000' '1995-10-11 00:00:00.000'
  '1995-10-12 00:00:00.000']
 ['1995-10-13 00:00:00.000' '1995-10-14 00:00:00.000'
  '1995-10-15 00:00:00.000']
 ['1995-10-16 00:00:00.000' '1995-10-17 00:00:00.000'
  '1995-10-18 00:00:00.000']]>
```

From a quick look, I think your PR will not work for this 2-dimensional case. My own sense would be that we need to use numpy's `array2string` infrastructure a bit better, but that is not without its own pitfalls.

In any case, I think before anything else we need to be sure exactly what the desired outcome is!

File: astropy/time/tests/test_mask.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def test_str():
     t = Time(['2000:001', '2000:002'])
     t[1] = np.ma.masked
     assert str(t) == "['2000:001:00:00:00.000' --]"
-    assert repr(t) == "<Time object: scale='utc' format='yday' value=['2000:001:00:00:00.000' --]>"
+    assert repr(t) == "<Time object: scale='utc' format='yday' value=[2000:001:00:00:00.000 ... 0.0]>"
```
Comment (mhvk) [inline astropy/time/tests/test_mask.py]: This change does not look right to me - the masked elements should remain as masked, not `0.0`

**Comment by taldcroft** [comment]
The underlying issue is that we delegate array repr to numpy (which is a good thing!), and that uses a default `threshold` of 1000 for the number of elements to print. (https://docs.scipy.org/doc/numpy/reference/generated/numpy.set_printoptions.html)

So we can use the `np.printoptions` context manager to locally reduce the `threshold` as shown below.  I'm not sure of what the right value for `threshold` would be, but probably something between 10 and 100.
```
In [15]: t = Time(np.arange(1000), format='cxcsec')

In [16]: with np.printoptions(threshold=10):
    ...:     print(repr(t))
    ...:     
<Time object: scale='tt' format='cxcsec' value=[  0.   1.   2. ... 997. 998. 999.]>

In [17]: t.shape = (10, 10, 10)

In [18]: with np.printoptions(threshold=10):
    ...:     print(repr(t))
    ...:     
<Time object: scale='tt' format='cxcsec' value=[[[  0.   1.   2. ...   7.   8.   9.]
  [ 10.  11.  12. ...  17.  18.  19.]
  [ 20.  21.  22. ...  27.  28.  29.]
  ...
  [ 70.  71.  72. ...  77.  78.  79.]
  [ 80.  81.  82. ...  87.  88.  89.]
  [ 90.  91.  92. ...  97.  98.  99.]]

 [[100. 101. 102. ... 107. 108. 109.]
  [110. 111. 112. ... 117. 118. 119.]
  [120. 121. 122. ... 127. 128. 129.]
  ...
  [170. 171. 172. ... 177. 178. 179.]
  [180. 181. 182. ... 187. 188. 189.]
  [190. 191. 192. ... 197. 198. 199.]]

 [[200. 201. 202. ... 207. 208. 209.]
  [210. 211. 212. ... 217. 218. 219.]
  [220. 221. 222. ... 227. 228. 229.]
  ...
  [270. 271. 272. ... 277. 278. 279.]
  [280. 281. 282. ... 287. 288. 289.]
  [290. 291. 292. ... 297. 298. 299.]]

 ...

 [[700. 701. 702. ... 707. 708. 709.]
  [710. 711. 712. ... 717. 718. 719.]
  [720. 721. 722. ... 727. 728. 729.]
  ...
  [770. 771. 772. ... 777. 778. 779.]
  [780. 781. 782. ... 787. 788. 789.]
  [790. 791. 792. ... 797. 798. 799.]]

 [[800. 801. 802. ... 807. 808. 809.]
  [810. 811. 812. ... 817. 818. 819.]
  [820. 821. 822. ... 827. 828. 829.]
  ...
  [870. 871. 872. ... 877. 878. 879.]
  [880. 881. 882. ... 887. 888. 889.]
  [890. 891. 892. ... 897. 898. 899.]]

 [[900. 901. 902. ... 907. 908. 909.]
  [910. 911. 912. ... 917. 918. 919.]
  [920. 921. 922. ... 927. 928. 929.]
  ...
  [970. 971. 972. ... 977. 978. 979.]
  [980. 981. 982. ... 987. 988. 989.]
  [990. 991. 992. ... 997. 998. 999.]]]>
```

**Comment by colathro** [comment]
> The underlying issue is that we delegate array repr to numpy (which is a good thing!), and that uses a default `threshold` of 1000 for the number of elements to print. (https://docs.scipy.org/doc/numpy/reference/generated/numpy.set_printoptions.html)
> 
> So we can use the `np.printoptions` context manager to locally reduce the `threshold` as shown below. I'm not sure of what the right value for `threshold` would be, but probably something between 10 and 100.
> 
> ```
> In [15]: t = Time(np.arange(1000), format='cxcsec')
> 
> In [16]: with np.printoptions(threshold=10):
>     ...:     print(repr(t))
>     ...:     
> <Time object: scale='tt' format='cxcsec' value=[  0.   1.   2. ... 997. 998. 999.]>
> 
> In [17]: t.shape = (10, 10, 10)
> 
> In [18]: with np.printoptions(threshold=10):
>     ...:     print(repr(t))
>     ...:     
> <Time object: scale='tt' format='cxcsec' value=[[[  0.   1.   2. ...   7.   8.   9.]
>   [ 10.  11.  12. ...  17.  18.  19.]
>   [ 20.  21.  22. ...  27.  28.  29.]
>   ...
>   [ 70.  71.  72. ...  77.  78.  79.]
>   [ 80.  81.  82. ...  87.  88.  89.]
>   [ 90.  91.  92. ...  97.  98.  99.]]
> 
>  [[100. 101. 102. ... 107. 108. 109.]
>   [110. 111. 112. ... 117. 118. 119.]
>   [120. 121. 122. ... 127. 128. 129.]
>   ...
>   [170. 171. 172. ... 177. 178. 179.]
>   [180. 181. 182. ... 187. 188. 189.]
>   [190. 191. 192. ... 197. 198. 199.]]
> 
>  [[200. 201. 202. ... 207. 208. 209.]
>   [210. 211. 212. ... 217. 218. 219.]
>   [220. 221. 222. ... 227. 228. 229.]
>   ...
>   [270. 271. 272. ... 277. 278. 279.]
>   [280. 281. 282. ... 287. 288. 289.]
>   [290. 291. 292. ... 297. 298. 299.]]
> 
>  ...
> 
>  [[700. 701. 702. ... 707. 708. 709.]
>   [710. 711. 712. ... 717. 718. 719.]
>   [720. 721. 722. ... 727. 728. 729.]
>   ...
>   [770. 771. 772. ... 777. 778. 779.]
>   [780. 781. 782. ... 787. 788. 789.]
>   [790. 791. 792. ... 797. 798. 799.]]
> 
>  [[800. 801. 802. ... 807. 808. 809.]
>   [810. 811. 812. ... 817. 818. 819.]
>   [820. 821. 822. ... 827. 828. 829.]
>   ...
>   [870. 871. 872. ... 877. 878. 879.]
>   [880. 881. 882. ... 887. 888. 889.]
>   [890. 891. 892. ... 897. 898. 899.]]
> 
>  [[900. 901. 902. ... 907. 908. 909.]
>   [910. 911. 912. ... 917. 918. 919.]
>   [920. 921. 922. ... 927. 928. 929.]
>   ...
>   [970. 971. 972. ... 977. 978. 979.]
>   [980. 981. 982. ... 987. 988. 989.]
>   [990. 991. 992. ... 997. 998. 999.]]]>
> ```

Good idea! This is much more straight forward, thanks for the suggestion.

I also think the decimal placement is a bit odd, when exact we get just a bunch of whitespace instead of 0's. This is also adjustable via np print options.

**Comment by mhvk** [comment]
Definitely cleaner to use `printoptions`! One thing with this is that obviously the user now looses the ability to control these two options themselves. In `Quantity.to_string`, a configuration item was introduced to solve this (for latex representation only) - not sure whether we want to do similarly here, as there is also not much use for a whole slew of configuration items that essentially are never used. But it makes me wonder about using `fixed` in particular - my tendency would be to only use it if the time is not a scalar.

Another question is whether one might also want to clean up the `__repr__` a bit in having the first line of values for arrays already be on a new line (i.e., respect the `linewidth` print option); this would mean explicitly typesetting the values with `array2string`.

p.s. Of course, this is one of those cases in real risk of bikeshedding, since how one represents things is very much a matter of taste...

**Comment by colathro** [comment]
> Definitely cleaner to use `printoptions`! One thing with this is that obviously the user now looses the ability to control these two options themselves. In `Quantity.to_string`, a configuration item was introduced to solve this (for latex representation only) - not sure whether we want to do similarly here, as there is also not much use for a whole slew of configuration items that essentially are never used. But it makes me wonder about using `fixed` in particular - my tendency would be to only use it if the time is not a scalar.
> 
> Another question is whether one might also want to clean up the `__repr__` a bit in having the first line of values for arrays already be on a new line (i.e., respect the `linewidth` print option); this would mean explicitly typesetting the values with `array2string`.
> 
> p.s. Of course, this is one of those cases in real risk of bikeshedding, since how one represents things is very much a matter of taste...

As this doesn't seem to be intended to be used as object serialization (json or otherwise), the most common use case could be just generic stdout things like to console or a log file for debugging. 

I personally would prefer the output (in the above use cases) not to respect any line width and never have newlines in the output. A single n-length line that summarizes the object, giving me the important bits and pieces I care about.

Is removing the line width limiting something I should pursue, or are we approaching marginal return development?

File: astropy/time/core.py [hunk 3]
```diff
@@ -674,9 +674,10 @@ def format(self, format):
         self._format = format
 
     def __repr__(self):
-        return ("<{} object: scale='{}' format='{}' value={}>"
+        with np.printoptions(threshold=10, floatmode='fixed'):
```
Comment (taldcroft) [inline astropy/time/core.py]: I would propose to set the threshold to 50, remove that `floatmode` setting, and call this done.  If somebody wants to have `floatmode='fixed'` formatting they can use a `np.printoptions` context manager around the repr statement.

File: astropy/time/core.py [hunk 4]
```diff
@@ -674,9 +674,10 @@ def format(self, format):
...
+            return ("<{} object: scale='{}' format='{}' value={}>"
                 .format(self.__class__.__name__, self.scale, self.format,
-                        getattr(self, self.format)))
+                    getattr(self, self.format)))
```
Comment (taldcroft) [inline astropy/time/core.py]: This needs to be aligned for PEP8 (basically revert this change).

File: astropy/time/tests/test_basic.py [hunk 5]
```diff
@@ -2102,3 +2102,10 @@ def test_ymdhms_output():
 def test_broadcasting_writeable():
     t = Time('J2015') + np.linspace(-1, 1, 10)*u.day
     t[2] = Time(58000, format="mjd")
+
+
+def test_repr_large_array():
+    t = Time(np.linspace(2458780, 2458790, 100), format='jd')
```
Comment (taldcroft) [inline astropy/time/tests/test_basic.py]: This test can be made a bit cleaner by using `t = Time(np.arange(100), format='mjd')`, which gives: `<Time object: scale='utc' format='mjd' value=[ 0.  1.  2. ... 97. 98. 99.]>`

File: CHANGES.rst (hunk 1 above)
Comment (taldcroft) [reply CHANGES.rst]: This should be more like:
```
  - ``Time`` now displays a maximum of 50 array values in the representation before triggering
    summarization. [#9525]
```

**Comment by taldcroft** [comment]
At the end of the day I think the best answer is to document how to control the repr formatting via `np.printoptions`.  This way the `Time` object behaves more like a numpy array and there is consistency in the repr API.
```
In [15]: t = Time(np.arange(1000), format='cxcsec')

In [16]: with np.printoptions(threshold=10):
    ...:     print(repr(t))
    ...:     
<Time object: scale='tt' format='cxcsec' value=[  0.   1.   2. ... 997. 998. 999.]>
```
@colathro - thanks for the contribution, sorry it didn't end up getting merged.

**Comment by colathro** [comment]
@taldcroft No, I'm sorry I didn't follow through with this! Regardless of the outcome, it was awesome to work with you guys at PyData.
